### PR TITLE
Feature/cu nda0hh onimaru state refactoring

### PR DESCRIPTION
**Pull request by devildrake**
Established enum for the states, so that when animations triggers are sent, instead of sending an integer we send a static_cast<int> of an enum, for more understandable code.
Established some code refactoring, virtual void OnDeath for Character class. Onimaru does an override for now.
Prepared code for abilities to be implemented (established some bools)
Established differentiation between CanShoot and Shooting (This is useful for state management)
Established Player abstract method OnAnimationFinished, that will be called from PlayerDeath script, either on Fang or on Onimaru, depending on who is active at the time.

### CODE REVIEW

**Comment by jcarlos0305** [comment]
Task linked: [CU-nda0hh Onimaru State Refactor](https://app.clickup.com/t/nda0hh)

**Review by GBMiro — DISMISSED** [review]

File: Penteract/Assets/Scripts/Character.cpp [hunk 1]
```diff
@@ -1,12 +1,19 @@
 #include "Character.h"
 
 void Character::GetHit(float damage_) {
+	bool wasAlive = lifePoints > 0.0f;
```
Comment (isaacfdez) [inline Penteract/Assets/Scripts/Character.cpp:4]: What is the purpose of this bool?

File: Penteract/Assets/Scripts/Player.h [hunk 2]
```diff
@@ -52,20 +54,24 @@ class Player : public Character {
...
+	bool canShoot = false;
 	float3 lookAtMousePlanePosition = float3(0, 0, 0);
 	ComponentCamera* lookAtMouseCameraComp = nullptr;
 	CameraController* cameraController = nullptr;
 	ComponentAudioSource* playerAudios[static_cast<int>(AudioPlayer::TOTAL)] = { nullptr };
 	float3 facePointDir = float3(0, 0, 0);
 	MovementDirection movementInputDirection = MovementDirection::NONE;
 	ComponentTransform* playerMainTransform = nullptr;
+	int ultimateChargePoints = 0;
+	int ultimateChargePointsTotal = 10;
```
Comment (Jordi-Gil) [inline Penteract/Assets/Scripts/Player.h:72]: If this value is not going to change in the whole gameplay, set it as constexpr

File: Penteract/Assets/Scripts/Character.cpp (hunk 1 above)
Comment (devildrake) [reply Penteract/Assets/Scripts/Character.cpp:4]: Modified to this commento:
	//We assume that the character is always alive when this method gets called, so no need to check if character was alive before taking lives

File: Penteract/Assets/Scripts/Character.cpp [hunk 3]
```diff
@@ -15,12 +22,10 @@ void Character::Recover(int recoveryLife_)
...
 
-void Character::SetTotalLifePoints(int totalLifePoints_)
-{
+void Character::SetTotalLifePoints(int totalLifePoints_) {
 	totalLifePoints = totalLifePoints_;
 }
 
-void Character::SetDamageHit(int damageHit_)
-{
+void Character::SetDamageHit(int damageHit_) {
```
[the remark this answers is not in the document]
Comment (devildrake) [reply Penteract/Assets/Scripts/Character.cpp:29]: Copy that

File: Penteract/Assets/Scripts/Character.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (devildrake) [reply Penteract/Assets/Scripts/Character.cpp:4]: As I said to isaac, this is substituted by a comment now.

File: Penteract/Assets/Scripts/Player.h (hunk 2 above)
Comment (devildrake) [reply Penteract/Assets/Scripts/Player.h:72]: using const, contexpr generates an error here.

**Review by Yonimevicio — APPROVED** [review]

**Review by GBMiro — DISMISSED** [review]

**Review by Jordi-Gil — APPROVED** [review]

**Pull request merged**